# Envs

In [1]:
# %pip install pyfirmata

Note: you may need to restart the kernel to use updated packages.


# Arduino board

In [ ]:
import pyfirmata
import time

board = pyfirmata.Arduino('COM8')

In [3]:
while True:
    print(f"HIGH")
    # For the relay board: HIGH = OFF
    board.digital[9].write(1) # digital pin 13 = built-in LED
    board.digital[10].write(1) # digital pin 13 = built-in LED
    time.sleep(10) # second
    print(f"LOW")
    # For the relay board: LOW = ON
    board.digital[9].write(0)
    board.digital[10].write(0)
    time.sleep(10)

HIGH


KeyboardInterrupt: 

# Keithley

In [5]:
import pyvisa

In [6]:
rm = pyvisa.ResourceManager('C:/windows/System32/visa64.dll')
print(rm.list_resources())

('USB0::0x05E6::0x2636::4480001::INSTR', 'ASRL8::INSTR')


In [1]:
from  keithley2600 import Keithley2600

In [2]:
k = Keithley2600('USB0::0x05E6::0x2636::4480001::INSTR', visa_library = 'C:/windows/System32/visa64.dll')

In [3]:
# turn on SMUA
k.smua.source.output = k.smua.OUTPUT_ON 

In [4]:
k.smua.source.output = k.smua.OUTPUT_OFF

## measurement of the gate resistance

In [27]:
# Restore 2600B defaults.
k.smua.reset()
# Select the current source function.
k.smua.source.func = k.smua.OUTPUT_DCAMPS
# Set the source range to 10 mA. = resolution
k.smua.source.rangei = 10e-3
# Set the current source to 10 mA. (Bias/ Dc)
k.smua.source.leveli = 10e-3
# Set the voltage limit to 10 V. = safety
k.smua.source.limitv = 10
# Enable 2-wire ohms.
k.smua.sense = k.smua.SENSE_LOCAL
# Set the voltage range to auto.
k.smua.measure.autorangev = k.smua.AUTORANGE_ON
# Turn on output.
k.smua.source.output = k.smua.OUTPUT_ON

In [28]:
# Retrieve a resistance reading.
import time
time.sleep(1) # for the system to be stable
measured_r = []
for i in range(0, 100): # make 100 measurement and take average
    sample_r = k.smua.measure.r()
    measured_r.append(sample_r)
    time.sleep(0.1)

In [29]:
# Turn off output.
k.smua.source.output = k.smua.OUTPUT_OFF

In [30]:
import numpy as np
print(f"r = {(sum(measured_r) / len(measured_r))/ 1e3} kOhm")

r = 327.66637 kOhm


REPORT OF THE MEASUREMENT 

- Detail: gate resistance of oect

  - 1. OECT 6: r = 423.90777 kOhm

  - 2. OECT 5: r = 211.31303 kOhm

  - 3. OECT 4: r = 327.66637 kOhm

